outputs.tf

```diff
@@ -1,79 +1,217 @@
 }
 
-output "public_subnets" {
-  description = "A map that contains all public subnets."
-  value       = aws_subnet.public
+output "intra_route_tables" {
+  description = "A map of intra route tables keyed by group. (aws_route_table)"
+  value       = try(aws_route_table.intra, null)
 }
 
-output "private_subnets" {
-  description = "A map that contains all private subnets."
-  value       = aws_subnet.private
+output "intra_route_table_associations" {
+  description = "A map of intra route table associations keyed by the subnets CIDR Blocks. (aws_route_table_association)"
+  value       = try(aws_route_table_association.intra, null)
 }
 
-output "intra_subnets" {
-  description = "A map that contains all intra subnets."
-  value       = aws_subnet.intra
+output "eips" {
+  description = "A map of Elastic IP Adresses (EIPs) keyed by availability zone. (aws_eip)"
+  value       = try(aws_eip.eip, null)
 }
 
 output "nat_gateways" {
-  description = "A map that contains all NAT Gateways."
-  value       = aws_nat_gateway.nat
+  description = "A map of NAT gatweways keyed by availability zone. (aws_nat_gateway)"
+  value       = try(aws_nat_gateway.nat_gateway, null)
+}
+
+output "private_route_tables" {
+  description = "A map of private route tables keyed by group. (aws_route_table)"
+  value       = try(aws_route_table.private, null)
+}
+
+output "private_route_table_associations" {
+  description = "A map of private route table associations keyed by the subnets CIDR Blocks. (aws_route_table_association)"
+  value       = try(aws_route_table_association.private, null)
+}
+
+output "routes_to_nat_gateways" {
+  description = "A map of routes to the NAT Gateways keyed by group. (aws_route)"
+  value       = try(aws_route.nat_gateway, null)
+}
+
+output "internet_gateway" {
+  description = "The Internet Gateway. (aws_internet_gateway)"
+  value       = try(aws_internet_gateway.internet_gateway[0], null)
+}
+
+output "public_route_tables" {
+  description = "A map of public route tables keyed by group. (aws_route_table)"
+  value       = try(aws_route_table.public, null)
+}
+
+output "public_route_table_associations" {
+  description = "A map of public route table associations keyed by the subnets CIDR Blocks. (aws_route_table_association)"
+  value       = try(aws_route_table_association.public, null)
+}
+
+output "routes_to_internet_gateway" {
+  description = "A map of routes to the Internet Gateway keyed by group. (aws_route)"
+  value       = try(aws_route.internet_gateway, null)
+}
+
+output "subnets" {
+  description = "A map of subnets keyed by CIDR Blocks. (aws_subnet)"
+  value       = try(aws_subnet.subnet, null)
+}
+
+# ----------------------------------------------------------------------------------------------------------------------
+# OUTPUT ALL INPUT VARIABLES
+# ----------------------------------------------------------------------------------------------------------------------
+
+output "module_inputs" {
```

Comment:
nice 👍  but we should have implemented and discuss it in the template before rolling out to our modules.

Reply:
yes, 100% agree..  but this is still pre-state... can change it later on.. wanted to see how this feels when implemented. already helped to namespace the inputs as naming would be weird with single outputs...